duco_api.py

```diff
@@ -21,6 +21,16 @@ def decode_response(rec):
     response = response.split(",")
     return response
 
+def get_duco_price():
+    """
+    A function for getting the current price of DUCO
+    """
+    api_response = get(API_URL)
+    if api_response.status_code == 200:
+        duco_price = round(float(api_response.json()["Duco price"]), 6)
+    else:
+        duco_price = .003
+    return duco_price
 
```

Comment:
don't forget the timer loop

> Get Duco Price
> starts a thread with a loop that runs every 15 seconds
> `duco_api.GetDucoPrice() # access the value by using the global variable <ducofiat>`

Reply:
Maybe add a getDucoPrice function with the loop and one without

Reply:
Yeah, separate ones will be better, but why you need one with a timer? I think if a developer wants the current DUCO price, they probably don't want it to be updated in the background. The developer expects a function to return the current price and finish execution. In my opinion, that implementation is enough. And I can't really see where your implementation with a timer would be useful. I could make a property of a class, which will have get_duco_price as a getter, but don't think that's necessary, because why not just call get_duco_price?

Reply:
you have a point. @revoxhere what is your opinion?

Reply:
Just remove the timer and add a return and it'll be a good function, no need to write essays here.

Reply:
if you setup the timer one to update a variable for a tkinter lable then you dont need another loop

```python
def GetDucoPrice(text=None):
    global ducofiat
    jsonapi = get("https://raw.githubusercontent.com/revoxhere/duco-statistics/master/api.json", data = None)
    if jsonapi.status_code == 200:
        content = jsonapi.content.decode()
        contentjson = loads(content)
        ducofiat = round(float(contentjson["Duco price"]), 6)
    else:
        ducofiat = .003
    if text != None:
        text.set(f"1 Duco = ${str(ducofiat)}  ")
    Timer(15, GetDucoPrice).start()
```

Reply:
But that's a bit specific to the wallet, it may be good to keep it though, why not make both. A threaded and a Single-Time return one.

Reply:
then add a var you can pass for the timer as well

```python
def GetDucoPrice(text=None, timer=10)
```

Reply:
That sounds good!

Reply:
I've made a function with the timer, please check